crates/storage/db-api/src/table.rs

```diff
@@ -100,6 +100,15 @@ pub trait Table: Send + Sync + Debug + 'static {
     type Value: Value;
 }
 
+/// Trait that provides object-safe access to the table's metadata.
+pub trait TableMetadata: Send + Sync + Debug + 'static {
+    /// The table's name.
+    fn name(&self) -> &'static str;
+
+    /// Whether the table is a `DUPSORT` table.
+    fn is_dupsort(&self) -> bool;
+}
```

Comment:
this is not needed, because we already have `trait Table`?

Reply:
i don't think it's possible because `Trait` has associated `const`. I get something like this:

```
error[E0038]: the trait `Table` cannot be made into an object
   --> crates/storage/db/src/tables/mod.rs:105:48
    |
105 |     fn tables() -> Box<dyn Iterator<Item = Box<dyn Table<Key = Box<dyn Encode>, Value = Box<dyn Encode>>>>>;
    |                                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `Table` cannot be made into an object
    |
note: for a trait to be "dyn-compatible" it needs to allow building a vtable to allow the call to be resolvable dynamically; for more information visit <https://doc.rust-lang.org/reference/items/traits.html#object-safety>
   --> /Users/alessandromazza/work/firewall/reth/crates/storage/db-api/src/table.rs:89:11
    |
89  |     const NAME: &'static str;
    |           ^^^^ the trait cannot be made into an object because it contains this associated `const`
...
92  |     const DUPSORT: bool;
    |           ^^^^^^^ the trait cannot be made into an object because it contains this associated `const`

```

Reply:
or am I missing something?

Reply:
I see, but then we should be able to impl this helper for all T: Table by default?

Reply:
you're right